Set4/28/line/next.cc

```diff
@@ -0,0 +1,14 @@
+#include "line.ih"
+
+string Line::next()
+{
+    if (d_position == string::npos)
+        return "";
+                                                        // Find next ws 
+    size_t endPosition = d_line.find_first_of(" \t", d_position);
+    string subString = d_line.substr(d_position, endPosition - d_position);
+                                                        // Cut next substring
+    d_position = d_line.find_first_not_of(" \t", endPosition);
+                                        // set d_pos to next non ws position
+    return subString;
+}
```

Comment:
```suggestion
#include "line.ih"

string Line::next()
{
    if (d_position == string::npos)
        return "";
                                                        // Find next ws 
    size_t endPosition = d_line.find_first_of(" \t", d_position);
    d_position = d_line.find_first_not_of(" \t", endPosition);
                                        // set d_pos to next non ws position
    return d_line.substr(d_position, endPosition - d_position);
                                                        // Cut next substring;
}
```

Simplified a bit more, as there's no need to assign the substring to a variable.

(If you do want to keep it, don't forget to `const` the substring :))

Reply:
I was meaning to condense that one a bit, yeah!

Reply:
I think this is crashing now since string::npos is used to actually select things inside a string this way! Will have to revert it I think

Reply:
We assign the substring so we can cut it out before setting d_pos to the next one (which might afterwards become npos!)